### ISSUE TITLE
pl.bolts breaks torchvision.datasets.MNIST download

### ISSUE THREAD

**Issue by murphyk** [PyTorchLightning/lightning-bolts#673]
After installing lightning-bolts (0.3.4) in google colab, I find that `mnist = torchvision.datasets.MNIST('data', download=True)` fails to download the data:
```
!pip install -q lightning-bolts
import pl_bolts
pl_bolts.__version__
import torch
import torchvision
mnist = torchvision.datasets.MNIST('data', download=True)

Downloading http://yann.lecun.com/exdb/mnist/https://ossci-datasets.s3.amazonaws.com/mnist/train-images-idx3-ubyte.gz
Failed to download (trying next):
HTTP Error 404: Not Found

Downloading https://ossci-datasets.s3.amazonaws.com/mnist/https://ossci-datasets.s3.amazonaws.com/mnist/train-images-idx3-ubyte.gz
Failed to download (trying next):
HTTP Error 404: Not Found
```
Similarly, `dm = MNISTDataModule(num_workers=0, data_dir='data')` fails to work.
(Note that the download URL for the lightning version is mangled by lightning.)
However, if I don't install lightning-bolts, then the download does work! 
```
import torch
import torchvision
mnist = torchvision.datasets.MNIST('data', download=True)
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz to data/MNIST/raw/train-images-idx3-ubyte.gz
```

Colab to reproduce the problem:
https://github.com/probml/pyprobml/blob/master/notebooks/mnist_lightning.ipynb

**Comment by Borda** [comment]
@akihironitta mind have look? probably connected to our fix #633

**Comment by akihironitta** [comment]
Hi @murphyk, thank you for reporting this issue. It seems like this bug happens with torchvision>=0.9.1. The fix will be included in the next release (0.3.5). Until then, a quick workaround is to run the following before you download:
```python
import torchvision
torchvision.datasets.MNIST.resources = [
    ("train-images-idx3-ubyte.gz", "f68b3c2dcbeaaa9fbdd348bbdeb94873"),
    ("train-labels-idx1-ubyte.gz", "d53e105ee54ea40749a09fcbcd1e9432"),
    ("t10k-images-idx3-ubyte.gz", "9fb629c4189551a2d022fa330f9573f3"),
    ("t10k-labels-idx1-ubyte.gz", "ec29112dd5afa0611ce80d1b7f02629c")
]
```
where this code snippet comes from torchvision's source code [here](https://github.com/pytorch/vision/blob/300a8a40132c5d3ee7dc936034c6046fd30ca4c0/torchvision/datasets/mnist.py#L38-L43).

---

@Borda Just marked the linked PR ready for review :] Could you have a look?
